web-app/src/common/pathfinding.spec.ts

```diff
@@ -0,0 +1,143 @@
+import expect from 'expect';
+import { parseGameStateFromMultilineString } from './utils';
+import { getWinnablePathCost } from './pathfinding';
+
+describe('Shortest path cost', () => {
+  it('Should return a valid winning path cost', () => {
+    const input = parseGameStateFromMultilineString(`
+  ⬡ ⬡ ⬡
+   ⬡ ⬡ ⬡
+    ⬢ ⬡ ⬡
+  `);
+
+    const cost = getWinnablePathCost(input, "black");
+
+    expect(cost).toStrictEqual(201);
+  });
+
+  it('Should return a valid winning path cost', () => {
+    const input = parseGameStateFromMultilineString(`
+  ⬡ ⬡ ⬡
+   ⬡ ⬡ ⬡
+    ⬢ ⬢ ⬡
+  `);
+
+    const cost = getWinnablePathCost(input, "black");
+
+    expect(cost).toStrictEqual(102);
+  });
+
+  it('Should return a valid winning path cost', () => {
+    const input = parseGameStateFromMultilineString(`
+  ⬡ ⬢ ⬢
+   ⬡ ⬡ ⬡
+    ⬡ ⬡ ⬢
+  `);
+
+    const cost = getWinnablePathCost(input, "black");
+
+    expect(cost).toStrictEqual(102);
+  });
+
+  it('Should return a valid winning path cost', () => {
+    const input = parseGameStateFromMultilineString(`
+  ⬢ ⬡ ⬢
+   ⬡ ⬡ ⬡
+    ⬡ ⬡ ⬡
+  `);
+
+    const cost = getWinnablePathCost(input, "black");
+
+    expect(cost).toStrictEqual(102);
+  });
+
+  it('Should return a valid winning path cost', () => {
+    const input = parseGameStateFromMultilineString(`
+  ⬢ ⬢ ⬡
+   ⬡ ⬡ ⬡
+    ⬡ ⬡ ⬡
+  `);
+
+    const cost = getWinnablePathCost(input, "black");
+
+    expect(cost).toStrictEqual(102);
+  });
+
+  it('Should return a valid winning path cost', () => {
+    const input = parseGameStateFromMultilineString(`
+  ⬡ ⬡ ⬡
+   ⬡ ⬡ ⬡
+    ⬡ ⬡ ⬡
+  `);
+
+    const cost = getWinnablePathCost(input, "white");
+
+    expect(cost).toStrictEqual(300);
+  });
+
+  it('Should return a valid winning path cost', () => {
+    const input = parseGameStateFromMultilineString(`
+  ⬢ ⬢ ⬢
+   ⬡ ⬡ ⬡
+    ⬡ ⬡ ⬡
+  `);
+
+    const cost = getWinnablePathCost(input, "black");
+
+    expect(cost).toStrictEqual(3);
+  });
+
+  it('Should return a valid winning path cost', () => {
+    const input = parseGameStateFromMultilineString(`
+  W ⬡ ⬡
+   ⬡ W ⬡
+    ⬡ ⬡ W
+  `);
+
+    const cost = getWinnablePathCost(input, "white");
+
+    expect(cost).toStrictEqual(3);
+  });

```

Comment:
I disagree with this test. To me, these white stones are **not** connected.

Reply:
Same mistake as here : https://github.com/marmelab/Hex/pull/84#discussion_r821708086